app/models/subscriber.rb

```diff
@@ -0,0 +1,32 @@
+class Subscriber
+  def self.for_discussion(discussion)
+    new(discussion)
+  end
+
+  def initialize(discussion)
+    @discussion = discussion
+    @subscriptions = Subscription.where(discussion_id: discussion.id)
+  end
+
+  def subscribe(users)
+    users.each do |user|
+      create_subscription_for_user(user)
+    end
+  end
+
+  private
+
+  attr_reader :discussion, :subscriptions
+
+  def create_subscription_for_user(user)
+    return if user_already_subscribed?(user)
+
+    Subscription.create(user: user, discussion: discussion)
+  end
+
+  def user_already_subscribed?(user)
+    subscriptions.any? do |subscription|
+      subscription.user_id == user.id
+    end
+  end
```

Comment:
doesn't this force you to load all the subscriptions and check them on the rails layer? you could just do another `where(user_id: user.id).count` making this a simple count query instead of loading 8 subscriptions unnecessarily.

Also, shouldn't this just be a uniqueness validation?